poollet/machinepoollet/mem/generic_test.go

```diff
@@ -0,0 +1,123 @@
+		*srv = *machine.NewFakeRuntimeService()
+		srv.SetMachineClasses([]*machine.FakeMachineClassStatus{
+			{
+				MachineClassStatus: iri.MachineClassStatus{
+					MachineClass: &iri.MachineClass{
+						Name: mc.Name,
+						Capabilities: &iri.MachineClassCapabilities{
+							CpuMillis:   mc.Capabilities.CPU().MilliValue(),
+							MemoryBytes: mc.Capabilities.Memory().Value(),
+						},
+					},
+				},
+			},
+		})
+
+		k8sManager, err := ctrl.NewManager(cfg, ctrl.Options{
+			Scheme: scheme.Scheme,
+			Metrics: metricserver.Options{
+				BindAddress: "0",
+			},
+		})
+		Expect(err).ToNot(HaveOccurred())
+
+		machineClassMapper := mcm.NewGeneric(srv, mcm.GenericOptions{
+			RelistPeriod: 2 * time.Second,
+		})
+		Expect(k8sManager.Add(machineClassMapper)).To(Succeed())
+
+		machineEventMapper = mem.NewGeneric(k8sManager.GetClient(), srv, &record.FakeRecorder{}, mem.GenericOptions{
+			RelistPeriod: 2 * time.Second,
+		})
+		Expect(k8sManager.Add(machineEventMapper)).To(Succeed())
+
+		Expect((&controllers.MachineReconciler{
+			EventRecorder:         &record.FakeRecorder{},
+			Client:                k8sManager.GetClient(),
+			MachineRuntime:        srv,
+			MachineRuntimeName:    machine.FakeRuntimeName,
+			MachineRuntimeVersion: machine.FakeVersion,
+			MachineClassMapper:    machineClassMapper,
+			MachinePoolName:       mp.Name,
+			DownwardAPILabels: map[string]string{
+				fooDownwardAPILabel: fmt.Sprintf("metadata.annotations['%s']", fooAnnotation),
+			},
+		}).SetupWithManager(k8sManager)).To(Succeed())
+
+		mgrCtx, cancel := context.WithCancel(context.Background())
+		DeferCleanup(cancel)
+
+		go func() {
+			defer GinkgoRecover()
+			Expect(k8sManager.Start(mgrCtx)).To(Succeed(), "failed to start manager")
+		}()
+	})
+
+	It("should get event list for machine", func(ctx SpecContext) {
+		By("creating a machine")
+		const fooAnnotationValue = "bar"
+		machine := &computev1alpha1.Machine{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace:    ns.Name,
+				GenerateName: "machine-",
+				Annotations: map[string]string{
+					fooAnnotation: fooAnnotationValue,
+				},
+			},
+			Spec: computev1alpha1.MachineSpec{
+				MachineClassRef: corev1.LocalObjectReference{Name: mc.Name},
+				MachinePoolRef:  &corev1.LocalObjectReference{Name: mp.Name},
+			},
+		}
+		Expect(k8sClient.Create(ctx, machine)).To(Succeed())
+		By("waiting for the runtime to report the machine, volume and network interface")
+		Eventually(srv).Should(SatisfyAll(
+			HaveField("Machines", HaveLen(1)),
+		))
+
+		By("getting mchine events")
+		Eventually(func(g Gomega) []*iri.Event {
+			resp, err := machineEventMapper.GetMachineEventFor(ctx, string(machine.GetUID()))
+			g.Expect(err).NotTo(HaveOccurred())
+			return resp
+		}).Should(ConsistOf(SatisfyAll(
+			HaveField("Spec", SatisfyAll(
+				HaveField("Reason", Equal("testing")),
+				HaveField("Message", Equal("this is test event")),
+				HaveField("Type", Equal(corev1.EventTypeNormal)),
+			)),
+		)))
```

Comment:
Can you please check in the `testenv` if the event was emitted?

Reply:
added check to see if event was emitted